### ISSUE TITLE
(redis@4.0.0-next.5) Multi.exec throws TypeError when Watched key changes

### ISSUE THREAD

**Issue by Dan-Homerick-on24** [NodeRedis/node-redis#1636]
### Issue

When a watched key changes Redis will return null. In multi.exec, the null causes this TypeError:

```
TypeError: Cannot read property 'map' of null
    at RedisMultiCommand.exec (/.../redis/dist/lib/multi-command.js:77:50)
        at processTicksAndRejections (node:internal/process/task_queues:94:5)
            ...
```

from line 2 of this block of code in multi-command.js:

```javascript
const rawReplies = await __classPrivateFieldGet(this, _RedisMultiCommand_executor, "f").call(this, queue, Symbol('[RedisMultiCommand] Chain ID'));
return rawReplies[rawReplies.length - 1].map((reply, i) => {
    const { transformReply, preservedArguments } = queue[i + 1];
    return transformReply ? transformReply(reply, preservedArguments) : reply;
});
```

My code that triggered the bug (slightly simplified) was:
```javascript
await this.client.watch(hashmap_id);
let existing_value = await this.client.hGet(hashmap_id, field_id_1);
let multi = await this.client.multi();
multi.hSet(hashmap_id, { field_id_1: value1, field_id_2 : value2 });
let multi_result = await multi.exec();
```

Between the watch and the exec, I changed a value in the `hashmap_id` hash map.

I have not checked whether this bug occurs with other data types than hash maps.

---

### Environment

 - **Node.js Version**: `v15.12.0`
 

 - **Redis Version**: `6.2.5`
 

 - **Platform**: `MacOS "Big Sur" 11.5.1`

 - **Node-Redis Version**: `v4.0.0-next.5`

**Comment by Dan-Homerick-on24** [comment]
Still throws a TypeError instead of a WatchError in the redis@4.0.0-next.6 release.

**Comment by leibale** [comment]
@Dan-Homerick-on24 fixed in `redis@4.0.0-next.7`
